core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintUtilV4Controller.java

```diff
@@ -104,6 +110,23 @@ public RecommendationV4Response createRecommendationByCredCrn(Long workspaceId,
         return platformRecommendationToPlatformRecommendationV4ResponseConverter.convert(recommendation);
     }
 
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.DESCRIBE_ENVIRONMENT)
+    public RecommendationV4Response createRecommendationByEnvCrn(Long workspaceId, String blueprintName,
+            @TenantAwareParam @ResourceCrn String environmentCrn, @NotEmpty String region, String platformVariant,
+            String availabilityZone, CdpResourceType cdpResourceType) {
+        Credential credential = credentialClientService.getByEnvironmentCrn(environmentCrn);
```

Comment:
I think it would be better to have the least possible business logic here and the credentialClientService call would be moved to the BlueprintService class.

Reply:
agree, will do